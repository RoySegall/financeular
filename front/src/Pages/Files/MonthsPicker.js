import {getMonthAndYearFromKey} from "./File.service";
import React from "react";

export default ({months, selectedMonth, setCurrentMonth, setCurrentPage, setInfoMode}) => <ul className="flex mt-auto w-full ">
  {months.map((month, index) => {
    const selectedClass = month === selectedMonth ? 'font-extrabold text-black' : '';
    return <li key={index} className={`p-2 ${selectedClass}`}>
      <a className={"cursor-pointer"} onClick={() => { setCurrentMonth(month); setCurrentPage(0); setInfoMode('expenses') }}>{getMonthAndYearFromKey(month)}</a>
    </li>
  })}
</ul>
