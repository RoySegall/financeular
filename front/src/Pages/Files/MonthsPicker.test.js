import React from "react";
import {elementShouldExists, mountComponent} from "../../Tests/Utils";
import MonthsPicker from "./MonthsPicker";
import renderer from "react-test-renderer";

const months = ['1_2019', '2_2019', '3_2019', '1_2020'];
const mockSetCurrentMonth = jest.fn();
const mockSetCurrentPage = jest.fn();
const mockInfoMode = jest.fn();

describe('Month picker', () => {

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('Testing the component months without a picked month', () => {
    const tree = renderer
      .create(<MonthsPicker months={months}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Testing the component months with a picked month', () => {
    const months = ['1_2019', '2_2019', '3_2019', '1_2020'];
    const tree = renderer
      .create(<MonthsPicker months={months} selectedMonth={"1_2019"}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Testing the clicking on a month', () => {
    const wrapper = mountComponent({
      component: <MonthsPicker
        months={months}
        selectedMonth={"1_2019"}
        setCurrentMonth={mockSetCurrentMonth}
        setCurrentPage={mockSetCurrentPage}
        setInfoMode={mockInfoMode}
      />
    });

    elementShouldExists(wrapper.find('.font-extrabold.text-black'), 1);

    const firstNotSelectedAnchor = wrapper.find('li')
      .not('.font-extrabold.text-black')
      .at(0)
      .find('a');

    firstNotSelectedAnchor.simulate('click');

    expect(mockSetCurrentMonth).toBeCalledWith('2_2019');
    expect(mockSetCurrentPage).toBeCalledWith(0);
    expect(mockInfoMode).toBeCalledWith('expenses');
  });

});