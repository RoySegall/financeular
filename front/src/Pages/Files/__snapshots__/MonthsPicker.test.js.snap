// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month picker Testing the component months with a picked month 1`] = `
<ul
  className="flex mt-auto w-full "
>
  <li
    className="p-2 font-extrabold text-black"
  >
    <a
      className="cursor-pointer"
      onClick={[Function]}
    >
      January 2019
    </a>
  </li>
  <li
    className="p-2 "
  >
    <a
      className="cursor-pointer"
      onClick={[Function]}
    >
      February 2019
    </a>
  </li>
  <li
    className="p-2 "
  >
    <a
      className="cursor-pointer"
      onClick={[Function]}
    >
      March 2019
    </a>
  </li>
  <li
    className="p-2 "
  >
    <a
      className="cursor-pointer"
      onClick={[Function]}
    >
      January 2020
    </a>
  </li>
</ul>
`;

exports[`Month picker Testing the component months without a picked month 1`] = `
<ul
  className="flex mt-auto w-full "
>
  <li
    className="p-2 "
  >
    <a
      className="cursor-pointer"
      onClick={[Function]}
    >
      January 2019
    </a>
  </li>
  <li
    className="p-2 "
  >
    <a
      className="cursor-pointer"
      onClick={[Function]}
    >
      February 2019
    </a>
  </li>
  <li
    className="p-2 "
  >
    <a
      className="cursor-pointer"
      onClick={[Function]}
    >
      March 2019
    </a>
  </li>
  <li
    className="p-2 "
  >
    <a
      className="cursor-pointer"
      onClick={[Function]}
    >
      January 2020
    </a>
  </li>
</ul>
`;
